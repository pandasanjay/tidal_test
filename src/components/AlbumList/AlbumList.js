import React, { useState } from "react";
const Album = ({ list = [], onChange = () => {} }) => {
  const [active, setActive] = useState("");
  const handelClick = id => {
    setActive(id);
    onChange(id);
  };
  return (
    <>
      <div className="flex-container scroll">
        {list.map((data, i) => (
          <div
            key={i}
            className={
              "flex text-center album-section " +
              (active === data.id ? "active" : "")
            }
            onClick={() => handelClick(data.id)}
          >
            <img src={data.cover} alt={data.title} />
            <div className="title-text-sm">{data.title}</div>
          </div>
        ))}
      </div>
    </>
  );
};
export default Album;
