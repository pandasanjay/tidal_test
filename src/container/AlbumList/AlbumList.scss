@import "../../config";
.search-result-album-section {
  padding: $padding;
  h4 {
    padding: $padding 0;
    border-bottom: 1px solid $separator-color;
  }
}
.scroll {
  flex-wrap: nowrap;
  overflow-x: scroll;
  overflow-y: hidden;
  -webkit-overflow-scrolling: touch;
  -ms-overflow-style: -ms-autohiding-scrollbar;
}
.scroll::-webkit-scrollbar {
  display: none;
}
.album-section {
  width: 125px;
  cursor: pointer;
  &:nth-child(1) {
    margin-left: 0px;
  }
  &:active,
  &.active {
    border-bottom: 1px solid $separator-color;
  }
}

.album-img {
  position: absolute;
  z-index: 1;
  width: 120px;
  height: 120px;
}
.hide-section {
  position: absolute;
  height: 95px;
  background: $primary-color;
  top: 0;
  left: 0;
  right: 0;
}
.field-section {
  width: 120px;
  position: relative;
  padding: $padding;
  background: $highlight-color;
}
