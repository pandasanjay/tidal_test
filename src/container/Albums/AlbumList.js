import React, { useState } from "react";
const Album = ({ list = [], onSelectAlbum = () => {} }) => {
  const [active, setActive] = useState("");
  const handelClick = id => {
    setActive(id);
    onSelectAlbum(id);
  };
  return (
    <>
      <div className="flex-container scroll">
        {list.map((data, i) => (
          <div
            key={i}
            className={
              "flex text-center album-section " +
              (active === data.id ? "active" : "")
            }
            onClick={() => handelClick(data.id)}
          >
            <img src={data.cover} alt={data.title} />
            <div className="title-text-sm">{data.title}</div>
          </div>
        ))}
      </div>
    </>
  );
};
export default Album;
