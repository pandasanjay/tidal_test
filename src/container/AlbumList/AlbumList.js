import React from "react";
import "./AlbumList.scss";
import Album from "./Album";
import AlbumDetails from "./AlbumDetails";
const AlbumList = props => {
  return (
    <>
      <div className="search-result-album-section">
        <h4>Search Results for "{props.appReducer.searchText}"</h4>
        <div className="title-text pad-t-10 pad-b-10">ALBUMS</div>
        <Album
          list={props.appReducer.albums}
          onSelectAlbum={props.findAlbumDetailsByArtist}
        />
        <AlbumDetails details={props.appReducer.albumsDetails} />
      </div>
    </>
  );
};
export default AlbumList;
