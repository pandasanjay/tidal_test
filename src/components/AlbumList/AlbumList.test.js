import React from "react";
import { render } from "@testing-library/react";
import AlbumList from "./AlbumList";
describe("AlbumDetails connected component ", () => {
  it("should render and check the snapshot", () => {
    const stubLists = [
      {
        id: "100",
        title: "foo",
        cover: "https://api.deezer.com/album/99754012/image"
      }
    ];
    const { asFragment } = render(<AlbumList lists={stubLists} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
