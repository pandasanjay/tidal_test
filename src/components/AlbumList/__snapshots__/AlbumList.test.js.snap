// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumDetails connected component  should render and check the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-container scroll"
  />
</DocumentFragment>
`;
